src/compiler/build/test/build-stats.spec.ts

```diff
@@ -16,7 +16,7 @@ describe('generateBuildStats', () => {
   it('should return a structured json object', async () => {
     buildCtx.buildResults = generateBuildResults(config, compilerCtx, buildCtx);
 
-    const result: d.CompilerBuildStats = generateBuildStats(config, buildCtx);
+    const result = generateBuildStats(config, buildCtx) as d.CompilerBuildStats;
```

Comment:
because of the two possible return types here I thought this was the easier way to get this passing (and fairly well-justified)

Reply:
sure, that makes sense - I'm also not a big fan of sort of random `TypeA | TypeB` return types on functions. It would be one thing if we had something like `Result` or `Option` from Rust, so we could do `Result<TypeA, TypeB>` and have some more principled way for dealing with that, but we don't so :/

Reply:
Created `STENCIL-461` for this